src/modules/apps/publish.ts

```diff
@@ -53,30 +53,46 @@ const promptPublishOnVendor = (msg: string): Bluebird<boolean> =>
 const publisher = (workspace: string = 'master') => {
 
   const publishApp = async (appRoot: string, appId: string, tag: string, builder): Promise<BuildResult> => {
-    const stickyHint = await getMostAvailableHost(
-      appId,
-      builder,
-      N_HOSTS,
-      AVAILABILITY_TIMEOUT
-    )
-    const publishOptions = {
-      sticky: true,
-      stickyHint,
-      tag,
-    }
 
     const paths = await listLocalFiles(appRoot)
     const filesWithContent = map(pathToFileObject(appRoot), paths)
     log.debug('Sending files:', '\n' + paths.join('\n'))
-    try {
-      return await builder.publishApp(appId, filesWithContent, publishOptions)
-    } catch (e) {
-      const data = e.response && e.response.data
-      if (data && data.code && data.code === 'build_in_progress') {
-        log.warn(`Build for ${appId} is already in progress`)
+    const retryOpts = {
+      retries: 2,
+      minTimeout: 1000,
+      factor: 2,
+    }
+    const publish = async (bail, tryCount) => {
+      if (tryCount > 1) {
+        log.info(`Retrying...${tryCount-1}`)
+      }
+      let publishOptions
+      const stickyHint = await getMostAvailableHost(
+        appId,
+        builder,
+        N_HOSTS,
+        AVAILABILITY_TIMEOUT
+      )
+      publishOptions = {
+        sticky: true,
+        stickyHint,
+        tag,
+      }
+      try {
+        return await builder.publishApp(appId, filesWithContent, publishOptions)
+      } catch (err) {
+        const data = err.response && err.response.data
+        if (data && data.code && data.code === 'build_in_progress') {
+          log.warn(`Build for ${appId} is already in progress`)
```

Comment:
nipicking: tryCount can be useful in all log messages, since toolbelt can print logs out of order.

Reply:
Good idea. Done